helmclient.go

```diff
@@ -67,13 +68,28 @@ func New(config Config) (*Client, error) {
 
 // DeleteRelease uninstalls a chart given its release name.
 func (c *Client) DeleteRelease(releaseName string, options ...helmclient.DeleteOption) error {
-	t, err := c.newTunnel()
-	if err != nil {
-		return microerror.Mask(err)
+	o := func() error {
+		t, err := c.newTunnel()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		defer c.closeTunnel(t)
+
+		_, err = c.newHelmClientFromTunnel(t).DeleteRelease(releaseName, options...)
+		if IsReleaseNotFound(err) {
+			return backoff.Permanent(releaseNotFoundError)
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
+	b := newExponentialBackoff(2*time.Minute, backoff.DefaultMaxInterval)
```

Comment:
I think it would be good to extract this `2*time.Minute` thing, as this is used in multiple places.

Reply:
How do you want it to look like?

Reply:
`var backOffMaxWait = 2 * time.Minute`

Reply:
I am going with this now. We can still sort this issue if you like. I am glad to provide a separate PR to improve the situation here. Just give me a shout.